src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -262,9 +264,24 @@ export const txHistoryApi = createApi({
 
                 currentCursor = cursor
 
-                // TODO(0xdef1cafe): check if data exists in store and stop paginating before upserting
-                // to reduce load on backend
-                dispatch(txHistory.actions.upsertTxs({ txs: transactions, accountId }))
+                const txState = (getState() as any).txHistory.txs as TxsState
```

Comment:
We might want to use our synthetic `State` type here to retain typing, which we use elsewhere to avoid the `ReduxState` circular dependency.

```suggestion
                const state: State = getState() as unknown as State // ReduxState causes circular dependency
                const txState = state.txHistory.txs
```

Which would require `src/state/apis/types.ts` to be updated accordingly:

```diff
diff --git a/src/state/apis/types.ts b/src/state/apis/types.ts
index ce96e3e7c..68419dd64 100644
--- a/src/state/apis/types.ts
+++ b/src/state/apis/types.ts
@@ -1,7 +1,9 @@
 import type { AssetsState } from 'state/slices/assetsSlice/assetsSlice'
 import type { Preferences } from 'state/slices/preferencesSlice/preferencesSlice'
+import type { TxHistory } from 'state/slices/txHistorySlice/txHistorySlice'

 export type State = {
   assets: AssetsState
   preferences: Preferences
+  txHistory: TxHistory
 }
```

Reply:
excellent suggestion - resolved in https://github.com/shapeshift/web/commit/7646c874c4a071e420e8a9ea5c995f908dcefc77